tests/spark-it/src/test/scala/org/apache/celeborn/tests/spark/CelebornFetchFailureSuite.scala

```diff
@@ -297,4 +297,45 @@ class CelebornFetchFailureSuite extends AnyFunSuite
       sparkSession.stop()
     }
   }
+
+  test("celeborn spark integration test - resubmit an unordered barrier stage") {
```

Comment:
This specific test will pass when we support barrier stages (even without the changes in this PR).
@RexXiong's changes did reproduce the issue - perhaps adapt it here ?